### PR TITLE
feat: use code interpreter as a tool, support internal_python

### PR DESCRIPTION
**Pull request by onemquan**
## Description

code exectuion as a tool.

## Motivation and Context

https://github.com/camel-ai/camel/issues/622
- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/code_execution.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self.verbose = verbose
+        if sandbox == "internal_python":
+            self.interpreter = InternalPythonInterpreter()
+        else:
+            raise RuntimeError(
+                f"The sandbox type `{sandbox}` is not supported."
+            )
+
+    def execute_code(self, code: str) -> str:
+        r"""execute a giving code snippet.
```
Comment (WHALEEYE) [inline camel/toolkits/code_execution.py:39]: Nit: small typo
```suggestion
        r"""Execute a given code snippet.
```

File: camel/toolkits/code_execution.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import List, Literal
+
+from camel.functions import OpenAIFunction
+from camel.interpreters import InternalPythonInterpreter
+
+from .base import BaseToolkit
+
+
+class CodeExecutionToolkit(BaseToolkit):
```
Comment (WHALEEYE) [inline camel/toolkits/code_execution.py:22]: Could use some docstrings here?

**Review by WHALEEYE — COMMENTED** [review]
I think `__init__.py` also need to be updated to export `CodeExecutionToolkit`. Besides this and some small doc problems commented, this PR basically LGTM.

**Review by WHALEEYE — APPROVED** [review]

**Review by willshang76 — APPROVED** [review]

**Review by Wendong-Fan — APPROVED** [review]
Thanks @onemquan 's contribution! LGTM

**Pull request merged**